Problem: debugging scripts is hard

yrashk opened the pull request:
There is no easy way to look into what script is doing to be
able to debug it.

Solution: add experimental TRACE instruction to pumpkindb-term
(not available outside of it) that allows users to trace any values.

```
PumpkinDB> "test" TRACE 1.
Trace: "test"
0x01
```

yrashk committed:
Problem: debugging scripts is hard

There is no easy way to look into what script is doing to be
able to debug it.

Solution: add experimental TRACE instruction to pumpkindb-term
(not available outside of it) that allows users to trace any values.

```
PumpkinDB> "test" TRACE 1.
Trace: "test"
0x01
```

Closes #227

stuarth reviewed (approved):
Looks good! No objection to `TRACE` as the name. Let's follow up with an issue to add docs (or expand this to include).

yrashk commented:
The reason any I didn't add docs yet is because it's not implemented in the
engine but term. But maybe the doc should just indicate the context?

stuarth commented:
A couple other options: term welcome text, term help text, README 'Trying it out' section. Someplace new users will bump into it 😄

yrashk committed:
Problem: TRACE instruction is hard to discover

Solution: add a documentation for it (mentioning the fact
that it is only defined in pumpkindb-term) and mention it
in pumpkindb-term's help message.

yrashk commented:
Added documentation
